include/openssl/core.h

```diff
@@ -229,6 +229,27 @@ typedef int (OSSL_PASSPHRASE_CALLBACK)(char *pass, size_t pass_size,
                                        size_t *pass_len,
                                        const OSSL_PARAM params[], void *arg);
 
+/*-
+ * Provider configuration parameters are normally set in the configuration file,
+ * but can also be set early in the main program before a provider is in use by
+ * multiple threads.  The "default" provider does not load any parameters from
+ * the configuration file and this is the only way to set "default" provider
+ * parameters.
+ *
+ * Only UTF8-string values are supported.
+ */
+int OSSL_PROVIDER_add_conf_parameter(OSSL_PROVIDER *prov, const char *name,
+                                     const char *value);
+/*
+ * Retrieves any of the requested configuration parameters for the given
+ * provider that were set in the configuration file or via the above
+ * OSSL_PROVIDER_add_parameter() function.
+ *
+ * The |params| array elements MUST have type OSSL_PARAM_UTF8_PTR, values are
+ * returned by reference, not as copies.
+ */
+int OSSL_PROVIDER_get_conf_parameters(OSSL_PROVIDER *prov, OSSL_PARAM params[]);
+
```

Comment:
Why not in include/openssl/provider.h?

Reply:
Not sure, can be moved if you think that'd  be better.

Reply:
Done.